lib/assets/javascripts/twine.js.coffee

```diff
@@ -194,7 +203,7 @@ wrapFunctionString = (code, args, node) ->
       ($context, $root) -> getValue($context, keypath)
   else
     try
-      new Function(args, "with($context) { return #{code} }")
+      new Function(args, "with($context) { with($modules) { return #{code} } }")
```

Comment:
Ok, did some benchmarks and read some articles:

### Performance pro & contra

__Pro:__ The `with` statement can help reduce file size by reducing the need to repeat a lengthy object reference without performance penalty. The scope chain change required by 'with' _is not computationally expensive_. Use of `with` will _relieve the interpreter_ of parsing repeated object references. Note, however, that in many cases this benefit can be achieved by using a temporary variable to store a reference to the desired object.

__Contra:__ The with statement forces the specified object to be searched first for all name lookups. Therefore all identifiers that aren't members of the specified object will be found more slowly in a `with` block. Where performance is important, `with` should only be used to encompass code blocks that access members of the specified object.

----

> `with` should only be used to encompass code blocks that access members of the specified object.

I think this is the biggest take away ^ So we should use `with` conditionally, SO when using `define` or `eval` it wraps the nested `with` which includes `$modules` otherwise it remains as is

Something like?

```coffee
"""
with ($context) {
  if (typeof $modules != 'undefined') {
    with ($modules) {
      return #{code}
    }
  } else {
    return #{code}
  }
}
"""
```

Any thoughts?